zzip/mmapped.c

```diff
@@ -275,8 +275,9 @@ zzip_disk_entry_to_data(ZZIP_DISK * disk, struct zzip_disk_entry * entry)
 struct zzip_file_header *
 zzip_disk_entry_to_file_header(ZZIP_DISK * disk, struct zzip_disk_entry *entry)
 {
-    zzip_byte_t *const ptr = disk->buffer + zzip_disk_entry_fileoffset(entry);
-    if (disk->buffer > ptr || ptr >= disk->endbuf)
+    zzip_off_t off = zzip_disk_entry_fileoffset(entry);
+    zzip_byte_t *const ptr = disk->buffer + off;
+    if (disk->buffer > ptr || disk->buffer >= disk->endbuf - off)
```

Comment:
Hey, I've tried the patch, however I was getting a segfault in valgrind, here're some gdb prints that seemed relevant:
```
(gdb) print disk->buffer > ptr
$12 = 0
(gdb) print (disk->buffer > ptr)
$13 = 0
(gdb) print (disk->buffer >= disk->endbuf - off)
$14 = 0
(gdb) print (ptr >= disk->endbuf)
$15 = 1
(gdb) print ptr
$16 = (zzip_byte_t * const) 0x58dad473 <error: Cannot access memory at address 0x58dad473>
(gdb) print disk->buffer
$17 = (zzip_byte_t *) 0x4856000 "PK\003\004\n"
(gdb) print disk->endbuf
$18 = (zzip_byte_t *) 0x485609b ""
(gdb) print off
$19 = 1414886515
(gdb) print disk->endbuf - off
$20 = (zzip_byte_t *) 0xffffffffb02fec28 <error: Cannot access memory at address 0xffffffffb02fec28>
```
There seems to be an issue where `off > disk->endbuff` is true. This underflows `disk->endbuf - off` and causes the second part of the condition to be false too. Something like this should probably help: 
`if (disk->buffer > ptr || disk->buffer >= disk->endbuf - off || disk->endbuf < off)`

Few notes:

1. I am trying to backport the patch to 0.13.68, so the main issue might not be completely relevant in the current master branch
2. there might some optimization that are causing troubles
3. I've used the input zip file from the referenced github issue

Reply:
Indeed, thanks for you message, I am closing this PR as a fix has been committed to the develop branch.